compiler/common-artifacts/src/TestDataGenerator.cpp

```diff
@@ -0,0 +1,234 @@
+  for (uint32_t i = 0; i < size; i++)
+  {
+    static_cast<T *>(data)[i] = static_cast<T>(distrib(gen));
+  }
+}
+
+void fill_random_data(void *data, uint32_t size, loco::DataType dtype)
+{
+  std::random_device rd;  // used to obtain a seed for the random number engine
+  std::mt19937 gen(rd()); // standard mersenne_twister_engine seeded with rd()
+
+  switch (dtype)
+  {
+    case loco::DataType::U8:
+      geneate_random_data<uint8_t>(gen, data, size);
+      break;
+    case loco::DataType::S32:
+      geneate_random_data<int32_t>(gen, data, size);
+      break;
+    case loco::DataType::S64:
+      geneate_random_data<int64_t>(gen, data, size);
+      break;
+    case loco::DataType::FLOAT32:
+      geneate_random_data<float>(gen, data, size);
+      break;
+    default:
+      break;
+  }
+}
+
+} // namespace
+
+int entry(int argc, char **argv)
+{
+  std::string circle_file{argv[1]};
+  size_t last_dot_index = circle_file.find_last_of(".");
+  std::string prefix = circle_file.substr(0, last_dot_index);
+
+  // load circle file
+  foder::FileLoader file_loader{circle_file};
+  std::vector<char> model_data = file_loader.load();
+  const circle::Model *circle_model = circle::GetModel(model_data.data());
+  if (circle_model == nullptr)
+  {
+    std::cerr << "ERROR: Failed to load circle '" << circle_file << "'" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // load luci module
+  std::unique_ptr<luci::Module> module = luci::Importer().importModule(circle_model);
+  luci_interpreter::Interpreter interpreter(module.get());
+
+  /**
+   *  HDF5 layout is like below
+   *
+   *  GROUP "/"
+   *   ㄴGROUP "name"
+   *     ㄴATTRIBUTE "0"
+   *       ㄴDATA (0): "input_01:0"
+   *     ㄴATTRIBUTE "1"
+   *       ㄴDATA (0): "input_02:0"
+   *   ㄴGROUP "value"
+   *     ㄴDATASET "0"
+   *       ㄴDATA ...
+   *     ㄴDATASET "1"
+   *       ㄴDATA ...
+   */
+  // create random data and dump into hdf5 file
+  H5::H5File input_file{prefix + ".input.h5", H5F_ACC_TRUNC};
+  std::unique_ptr<H5::Group> input_name_group =
+      std::make_unique<H5::Group>(input_file.createGroup("name"));
+  std::unique_ptr<H5::Group> input_value_group =
+      std::make_unique<H5::Group>(input_file.createGroup("value"));
+
+  H5::H5File output_file{prefix + ".expected.h5", H5F_ACC_TRUNC};
+  std::unique_ptr<H5::Group> output_name_group =
+      std::make_unique<H5::Group>(output_file.createGroup("name"));
+  std::unique_ptr<H5::Group> output_value_group =
+      std::make_unique<H5::Group>(output_file.createGroup("value"));
+
+  uint32_t input_index = 0;
+  for (uint32_t g = 0; g < circle_model->subgraphs()->size(); g++)
+  {
+    const auto input_nodes = loco::input_nodes(module->graph(g));
+    for (const auto &node : input_nodes)
+    {
+      const auto *input_node = dynamic_cast<const luci::CircleInput *>(node);
+      std::string name = input_node->name();
+      if (name.find(":0") == std::string::npos)
+        name += ":0";
```

Comment:
Is this a convention of OneRT ? This will add ":0" even if the name has ":1" or ":2".

Reply:
Good point! it's my mistake.